src/Core/AdminConsole/OrganizationFeatures/Policies/Enforcement/AutoConfirm/AutomaticUserConfirmationPolicyEnforcementValidator.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.AutoConfirmUser;
+using Bit.Core.AdminConsole.OrganizationFeatures.Policies.PolicyRequirements;
+using Bit.Core.AdminConsole.Repositories;
+using Bit.Core.AdminConsole.Utilities.v2.Validation;
+using static Bit.Core.AdminConsole.Utilities.v2.Validation.ValidationResultHelpers;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Policies.Enforcement.AutoConfirm;
+
+public class AutomaticUserConfirmationPolicyEnforcementValidator(
+    IPolicyRequirementQuery policyRequirementQuery,
+    IProviderUserRepository providerUserRepository)
+    : IAutomaticUserConfirmationPolicyEnforcementValidator
+{
+    public async Task<ValidationResult<AutomaticUserConfirmationPolicyEnforcementRequest>> IsCompliantAsync(
+        AutomaticUserConfirmationPolicyEnforcementRequest request)
+    {
+        var automaticUserConfirmationPolicyRequirement = await policyRequirementQuery
+            .GetAsync<AutomaticUserConfirmationPolicyRequirement>(request.User.Id);
+
+        var currentOrganizationUser = request.AllOrganizationUsers
+            .FirstOrDefault(x => x.OrganizationId == request.OrganizationId
+                                 && x.UserId == request.User.Id);
+
+        if (currentOrganizationUser is null)
+        {
+            return Invalid(request, new CurrentOrganizationUserIsNotPresentInRequest());
+        }
+
+        if ((await providerUserRepository.GetManyByUserAsync(request.User.Id)).Count != 0)
+        {
+            return Invalid(request, new ProviderUsersCannotJoin());
+        }
+
+        if (request.AllOrganizationUsers.Count == 1)
+        {
+            return Valid(request);
+        }
```

Comment:
You're missing a check that the policy applies to this user for the current org. e.g. `automaticUserConfirmationPolicyRequirement.IsEnabled(request.OrganizationId)`. (bonus points if you can get away from the `IsEnabled` wording but whatever works.)

As it is currently written these requirements will be enforced whether or not the policy is enabled.

Reply:
For this one, I think we do actually mean IsEnabled